### PR TITLE
feat: change loader component

### PR DESCRIPTION
**Pull request by MentorPK**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change

- [x]  Change [react-loader](https://www.npmjs.com/package/react-loader) to [react-loader-spinner](https://mhnpd.github.io/react-loader-spinner/). 
- [x] Remove old dependency
- [x] Adjust old design to new loader
- [x] Remove unused code

#### Motivation and Context

 Suggesting this PR to swap out deprecated lib ```react-loader``` regarding issue #1284 

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by mcmxcdev** [comment]
This is how the new loading indicator looks like:

[Screencast from 2023-10-20 09-57-42.webm](https://github.com/ferdium/ferdium-app/assets/16797721/34a93af9-32d8-4791-a4b3-38903d573933)

File: src/features/workspaces/components/WorkspaceSwitchingIndicator.tsx [hunk 1]
```diff
@@ -41,25 +40,23 @@ const styles = theme => ({
...
   spinner: {
-    width: 40,
-    height: 40,
     marginRight: 10,
   },
   message: {
     fontSize: 16,
     whiteSpace: 'nowrap',
-    color: theme.colorAppLoaderSpinner,
+    color: '#FFFFFF',
```
Comment (SpecialAro) [inline src/features/workspaces/components/WorkspaceSwitchingIndicator.tsx]: What does this change exactly? And why is it needed?

File: src/components/ui/loader/index.tsx [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+@inject('stores')
 class LoaderComponent extends Component<IProps> {
   render() {
-    const { classes, className, color } = this.props;
-
+    const {
+      classes,
+      className,
+      size = 36,
+      color = '#FFF',
```
Comment (SpecialAro) [inline src/components/ui/loader/index.tsx]: Can we maybe set the default to be 'primary' -> settings.app.accentColor ? That way we can remove all `<Loader color='primary' />` to `<Loader />`. What do you think?

**Review by SpecialAro — COMMENTED** [review]
Hi! I still haven't tested but overall LGTM! Great work. I left some comments 😄

File: src/components/ui/loader/index.tsx (hunk 2 above)
Comment (mcmxcdev) [reply src/components/ui/loader/index.tsx]: Well, we have to either pass white in to certain loaders if white ain't the default prop, or pass primary if white is the default, up to you!

File: src/features/workspaces/components/WorkspaceSwitchingIndicator.tsx (hunk 1 above)
Comment (mcmxcdev) [reply src/features/workspaces/components/WorkspaceSwitchingIndicator.tsx]: I wanted to get rid of some theme colors which are only used once, but looking at it, hardcoded white is also bad. Will undo.

Comment (mcmxcdev) [reply src/features/workspaces/components/WorkspaceSwitchingIndicator.tsx]: Undid the change in [`b4d3dd6` (#1410)](https://github.com/ferdium/ferdium-app/pull/1410/commits/b4d3dd6cb6ca30ff6db8a0a866339e1ba9d95683)

File: src/components/ui/loader/index.tsx (hunk 2 above)
Comment (mcmxcdev) [reply src/components/ui/loader/index.tsx]: Changed default color to be accentColor and now passing white color in as prop when needed: [`b4d3dd6` (#1410)](https://github.com/ferdium/ferdium-app/pull/1410/commits/b4d3dd6cb6ca30ff6db8a0a866339e1ba9d95683)

**Review by SpecialAro — APPROVED** [review]
LGTM! Thank you @MentorPK and @mcmxcdev

**Comment by SpecialAro** [comment]
@all-contributors please add @MentorPK for code.

**Pull request merged**